lib/ukboot/early_init.c

```diff
@@ -0,0 +1,128 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/argparse.h>
+#include <uk/assert.h>
+#include <uk/boot/earlytab.h>
+#include <uk/essentials.h>
+#include <uk/plat/common/bootinfo.h>
+#include <uk/prio.h>
+
+#ifdef CONFIG_LIBUKLIBPARAM
+#include <uk/libparam.h>
+#endif /* CONFIG_LIBUKLIBPARAM */
+
+char *argv[CONFIG_LIBUKBOOT_MAXNBARGS];
+int argc;
+
+static char cmdline[CONFIG_LIBUKBOOT_CMDLINE_MAX_LEN];
+static __sz cmdline_len;
+
+#ifdef CONFIG_LIBUKLIBPARAM
+static int init_libparam(int argc, char *argv[])
+{
+	int rc;
+	/*
+	 * First, we scan if we can find the stop sequence in the kernel
+	 * cmdline. If not, we assume that there are no uklibparam arguments in
+	 * the command line.
+	 * NOTE: argv[0] contains the kernel/program name that we need to
+	 *       hide from the parser.
+	 */
+	rc = uk_libparam_parse(argc - 1, &argv[1], UK_LIBPARAM_F_SCAN);
```

Comment:
Nitpick: perhaps `rc` is not really the proper name for the variable. Maybe `arg_idx`?

Reply:
Again, that's migrated from `boot.c`.